### ISSUE TITLE
Enable specifying specific input/output file data types

### ISSUE THREAD

**Issue by ValHayot** [boutiques/boutiques#217]
For an added layer of validation
e.g. NIfTI file, MINC file,...

**Comment by glatard** [comment]
Could/should be done by adding properties to inputs and outputs (similar to properties on tools).

**Comment by glatard** [comment]
Agreed at the May 3rd meeting. New property will be `file-extension`.

**Comment by ramou** [comment]
Would it be possible to extend this idea to specify file format. My preference would be to support a file_format regular expression. I'd originally considered line_format, but I think applying it broadly is flexible, and is no more technically difficult.

I think our layer of validation, where such things would happen, would exist after the jsonschema validation, but I haven't looked deeply into what other validation is happening with invocations and where it lives, so there my be a natural fit somewhere already.

**Comment by glatard** [comment]
Do you mean file *name* format, or really file format? Defining file formats isn't easy, and we had concluded in a previous meeting that file extensions were a good common denominator across users.
